### ISSUE TITLE
Harpoon Rope Crash

### ISSUE THREAD

**Issue by Hifume** [Regalis11/Barotrauma#9716]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

Game crashed after firing an Ahab's Spear at a Tiger Thresher in a Beacon. Appears to be that the harpoon rope was destroyed before it was done being used. Unknown if this is exclusive to Ahab's Spear, or Tiger Thresher. This happened during a campaign, although I doubt campaign was related to the bug. All content packages active were submarines. The currently loaded submarine was a mod sub ([SP] Bullet). Attached crash report is from the server host, after I fired a spear. My game did not crash.

### Reproduction steps

1. Start a server
2. Have a client load a Harpoon Gun with Ahab's Spears.
3. Have client fire a spear at a Tiger Thresher
4. Host game/server crashes.

### Bug prevalence

Just once

### Version

0.18.15.0

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

```shell
Barotrauma Client crash report (generated on 8/12/2022 3:16:25 AM)


Barotrauma seems to have crashed. Sorry for the inconvenience! 


116B6C514DB831BB68D5A8F96ACE87C4


Game version 0.18.15.1 (ReleaseWindows, branch hf, revision 91a05f31ff)
Graphics mode: 1920x1080 (BorderlessWindowed)
VSync ON
Language: English
Selected content packages: Vanilla, all doors, mogus, H, always flooding, Spider's Plan, Please Do Not The Sub., KS-09 Banan-2, fastmogus, mogus, Wait for the dentist, Sweet Pea Mk II, Sosig, [SP] Bullet, [SP] Bullet, SMS Olympus, fastmogus, Kessel, unholy., beeg hammer, beeg horn, big dugong, Magic (suicidal) capsule 5000, The Nomad, [VH] The Iroh, clown moment, clown moment 2, dysfunctional ballasts, Engineer Locker, Fabrication Cabinet, H, hmammer, Humpback (Deep diver), Humpback but BAD, No Walls, no water, Oxygen Tank Shelf, plants, perhaps, Supply Cabinet, various tests, Tringle
Level seed: Cadmus LineaCylymala
Loaded submarine: [SP] Bullet (F9BF852EF580C1A6004DA8D0B56015DB)
Selected screen: Barotrauma.GameScreen
SteamManager initialized
Client (Round had started)


System info:
    Operating system: Microsoft Windows NT 10.0.19044.0 64 bit
    GPU name: NVIDIA GeForce GTX 1060 6GB
    Display mode: {Width:1920 Height:1080 Format:Color AspectRatio:1.7777778}
    GPU status: Normal


Exception: Object reference not set to an instance of an object. (System.NullReferenceException)
Target site: Void Update(Single, Barotrauma.Camera)
Stack trace: 
   at Barotrauma.Items.Components.Rope.Update(Single deltaTime, Camera cam) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Items\Components\Rope.cs:line 0
   at Barotrauma.Item.Update(Single deltaTime, Camera cam) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Items\Item.cs:line 1848
   at Barotrauma.MapEntity.UpdateAll(Single deltaTime, Camera cam) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Map\MapEntity.cs:line 610
   at Barotrauma.GameScreen.Update(Double deltaTime) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Screens\GameScreen.cs:line 252
   at Barotrauma.GameMain.Update(GameTime gameTime) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\GameMain.cs:line 874
   at Microsoft.Xna.Framework.Game.DoUpdate(GameTime gameTime) in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 656
   at Microsoft.Xna.Framework.Game.Tick() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 500
   at Microsoft.Xna.Framework.SdlGamePlatform.RunLoop() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\SDL\SDLGamePlatform.cs:line 92
   at Microsoft.Xna.Framework.Game.Run(GameRunBehavior runBehavior) in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 397
   at Microsoft.Xna.Framework.Game.Run() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 367
   at Barotrauma.Program.Main(String[] args) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Program.cs:line 58


Last debug messages:
[8/12/2022 3:10:10 AM] Limb (RightFoot) angular velocity invalid (1031.7174, character: Swarmfeeder). Ragdoll controlled remotely. Playing as a client.
[8/12/2022 3:08:20 AM] Server running slowly (58 updates/s)!
[8/12/2022 2:51:24 AM] Received invalid SetAttackTarget message. Target entity not found (ID 0)
[8/12/2022 2:51:21 AM] Received invalid SetAttackTarget message. Target entity not found (ID 0)
[8/12/2022 2:50:01 AM] Server running slowly (58 updates/s)!
[8/12/2022 2:43:33 AM] Stan
[8/12/2022 2:43:33 AM] Beef!
[8/12/2022 2:43:33 AM] Ian
[8/12/2022 2:43:33 AM] Robbysan
[8/12/2022 2:43:32 AM] Stan
[8/12/2022 2:43:32 AM] Beef!
[8/12/2022 2:43:32 AM] Ian
[8/12/2022 2:43:32 AM] Robbysan
[8/12/2022 2:43:31 AM] Robbysan
[8/12/2022 2:43:28 AM] Received STARTGAMEFINALIZE packet. Round init status: Started
[8/12/2022 2:43:24 AM]  -1: WaitForPingDataAsync (owner), RanToCompletion
[8/12/2022 2:43:24 AM]  -0: WaitForPingDataAsync (owner), RanToCompletion
[8/12/2022 2:43:24 AM] Task count: 2
[8/12/2022 2:43:24 AM] Received STARTGAMEFINALIZE packet. Round init status: WaitingForStartGameFinalize
[8/12/2022 2:43:20 AM] Generating an outpost for the start of the level... (Location: Cadmus Linea Mine, level type: LocationConnection)
[8/12/2022 2:43:16 AM] Level identifier: greatseabasic
[8/12/2022 2:42:52 AM] Server running slowly (57 updates/s)!
[8/12/2022 2:42:50 AM] Stan
[8/12/2022 2:42:50 AM] Beef!
[8/12/2022 2:42:50 AM] Ian
[8/12/2022 2:42:50 AM] Robbysan
[8/12/2022 2:32:36 AM] Server running slowly (58 updates/s)!
[8/12/2022 2:26:09 AM] Stan
[8/12/2022 2:26:09 AM] Beef!
[8/12/2022 2:26:09 AM] Ian
[8/12/2022 2:26:09 AM] Robbysan
[8/12/2022 2:26:05 AM] Stan
[8/12/2022 2:26:05 AM] Beef!
[8/12/2022 2:26:05 AM] Ian
[8/12/2022 2:26:05 AM] Robbysan
[8/12/2022 2:26:04 AM] Robbysan
[8/12/2022 2:26:01 AM] Received STARTGAMEFINALIZE packet. Round init status: Started
[8/12/2022 2:25:57 AM]  -1: WaitForPingDataAsync (owner), RanToCompletion
[8/12/2022 2:25:57 AM]  -0: WaitForPingDataAsync (owner), RanToCompletion
[8/12/2022 2:25:57 AM] Task count: 2
[8/12/2022 2:25:57 AM] Received STARTGAMEFINALIZE packet. Round init status: WaitingForStartGameFinalize
[8/12/2022 2:25:54 AM] Generating an outpost for the start of the level... (Location: Cadmus Linea Mine, level type: LocationConnection)
[8/12/2022 2:25:50 AM] Level identifier: greatseabasic
[8/12/2022 2:25:28 AM] Stan
[8/12/2022 2:25:28 AM] Beef!
[8/12/2022 2:25:28 AM] Ian
[8/12/2022 2:25:28 AM] Robbysan
[8/12/2022 2:25:28 AM] Server running slowly (58 updates/s)!
[8/12/2022 2:25:10 AM] Stan
[8/12/2022 2:25:10 AM] Beef!
[8/12/2022 2:25:10 AM] Ian
[8/12/2022 2:25:10 AM] Robbysan
[8/12/2022 2:25:10 AM] Stan
[8/12/2022 2:25:10 AM] Beef!
[8/12/2022 2:25:10 AM] Ian
[8/12/2022 2:25:10 AM] Robbysan
[8/12/2022 2:25:10 AM] Stan
[8/12/2022 2:25:10 AM] Beef!
[8/12/2022 2:25:10 AM] Ian
[8/12/2022 2:25:10 AM] Robbysan
[8/12/2022 2:25:09 AM] Stan
[8/12/2022 2:25:09 AM] Beef!
[8/12/2022 2:25:09 AM] Ian
[8/12/2022 2:25:08 AM] Robbysan
[8/12/2022 2:25:07 AM] Received STARTGAMEFINALIZE packet. Round init status: Started
[8/12/2022 2:25:04 AM]  -1: WaitForPingDataAsync (owner), RanToCompletion
[8/12/2022 2:25:04 AM]  -0: WaitForPingDataAsync (owner), RanToCompletion
[8/12/2022 2:25:04 AM] Task count: 2
[8/12/2022 2:25:04 AM] Received STARTGAMEFINALIZE packet. Round init status: WaitingForStartGameFinalize
[8/12/2022 2:25:01 AM] Generating an outpost for the start of the level... (Location: Cadmus Linea Mine, level type: LocationConnection)
[8/12/2022 2:24:55 AM] Level identifier: greatseabasic
[8/12/2022 2:24:48 AM] Server running slowly (58 updates/s)!
[8/12/2022 2:24:37 AM] Server running slowly (57 updates/s)!
[8/12/2022 2:24:34 AM] Stan
[8/12/2022 2:24:34 AM] Beef!
[8/12/2022 2:24:34 AM] Ian
[8/12/2022 2:24:34 AM] Robbysan
[8/12/2022 2:24:34 AM] Stan
[8/12/2022 2:24:34 AM] Beef!
[8/12/2022 2:24:34 AM] Ian
[8/12/2022 2:24:34 AM] Robbysan
[8/12/2022 2:23:26 AM] Server running slowly (56 updates/s)!
[8/12/2022 2:23:23 AM] Stan
[8/12/2022 2:23:23 AM] Beef!
[8/12/2022 2:23:23 AM] Ian
[8/12/2022 2:23:23 AM] Robbysan
[8/12/2022 2:23:21 AM] Stan
[8/12/2022 2:23:21 AM] Beef!
[8/12/2022 2:23:21 AM] Ian
[8/12/2022 2:23:20 AM] Robbysan
[8/12/2022 2:23:17 AM] Received STARTGAMEFINALIZE packet. Round init status: Started
[8/12/2022 2:23:14 AM] Task count: 0
[8/12/2022 2:23:14 AM] Received STARTGAMEFINALIZE packet. Round init status: WaitingForStartGameFinalize
[8/12/2022 2:23:10 AM] Generating an outpost for the start of the level... (Location: Cadmus Linea Mine, level type: LocationConnection)
[8/12/2022 2:23:06 AM] Level identifier: greatseabasic
[8/12/2022 2:22:57 AM] Stan
[8/12/2022 2:22:57 AM] Beef!
[8/12/2022 2:22:57 AM] Ian
[8/12/2022 2:22:57 AM] Robbysan
[8/12/2022 2:22:57 AM] Stan
[8/12/2022 2:22:57 AM] Beef!
[8/12/2022 2:22:57 AM] Ian
[8/12/2022 2:22:57 AM] Robbysan
[8/12/2022 2:22:57 AM] Stan
[8/12/2022 2:22:57 AM] Beef!
[8/12/2022 2:22:57 AM] Ian
[8/12/2022 2:22:57 AM] Robbysan
[8/12/2022 2:22:57 AM] Stan
[8/12/2022 2:22:57 AM] Beef!
[8/12/2022 2:22:57 AM] Ian
[8/12/2022 2:22:57 AM] Robbysan
[8/12/2022 2:22:57 AM] Stan
[8/12/2022 2:22:57 AM] Beef!
[8/12/2022 2:22:57 AM] Ian
[8/12/2022 2:22:57 AM] Robbysan
[8/12/2022 2:22:56 AM] Stan
[8/12/2022 2:22:56 AM] Beef!
[8/12/2022 2:22:56 AM] Ian
[8/12/2022 2:22:56 AM] Robbysan
[8/12/2022 2:22:56 AM] Stan
[8/12/2022 2:22:56 AM] Beef!
[8/12/2022 2:22:56 AM] Ian
[8/12/2022 2:22:56 AM] Robbysan
[8/12/2022 2:22:56 AM] Stan
[8/12/2022 2:22:56 AM] Beef!
[8/12/2022 2:22:56 AM] Ian
[8/12/2022 2:22:56 AM] Robbysan
[8/12/2022 2:22:56 AM] Stan
[8/12/2022 2:22:56 AM] Beef!
[8/12/2022 2:22:56 AM] Ian
[8/12/2022 2:22:56 AM] Robbysan
[8/12/2022 2:22:55 AM] Stan
[8/12/2022 2:22:55 AM] Beef!
[8/12/2022 2:22:55 AM] Ian
[8/12/2022 2:22:55 AM] Robbysan
[8/12/2022 2:21:53 AM] Stan
[8/12/2022 2:21:53 AM] Beef!
[8/12/2022 2:21:53 AM] Ian
[8/12/2022 2:21:53 AM] Robbysan
[8/12/2022 2:21:52 AM] Beef!
[8/12/2022 2:21:52 AM] Ian
[8/12/2022 2:21:52 AM] Robbysan
[8/12/2022 2:21:38 AM] Beef!
[8/12/2022 2:21:38 AM] Ian
[8/12/2022 2:21:38 AM] Robbysan
[8/12/2022 2:21:37 AM] Beef!
[8/12/2022 2:21:37 AM] Ian
[8/12/2022 2:21:37 AM] Robbysan
[8/12/2022 2:21:37 AM] Beef!
[8/12/2022 2:21:37 AM] Ian
[8/12/2022 2:21:37 AM] Robbysan
[8/12/2022 2:21:36 AM] Beef!
[8/12/2022 2:21:36 AM] Ian
[8/12/2022 2:21:36 AM] Robbysan
[8/12/2022 2:20:38 AM] Beef!
[8/12/2022 2:20:38 AM] Ian
[8/12/2022 2:20:38 AM] Robbysan
[8/12/2022 2:16:45 AM] Server running slowly (58 updates/s)!
[8/12/2022 2:16:28 AM] Beef!
[8/12/2022 2:16:28 AM] Ian
[8/12/2022 2:16:28 AM] Stan
[8/12/2022 2:16:28 AM] Robbysan
[8/12/2022 2:11:28 AM] Ian
[8/12/2022 2:11:28 AM] Stan
[8/12/2022 2:11:28 AM] Robbysan
[8/12/2022 2:11:09 AM] Ian
[8/12/2022 2:11:09 AM] Stan
[8/12/2022 2:11:09 AM] Robbysan
[8/12/2022 2:11:06 AM] Ian
[8/12/2022 2:11:06 AM] Stan
[8/12/2022 2:11:06 AM] Robbysan
[8/12/2022 2:11:06 AM] Ian
[8/12/2022 2:11:06 AM] Stan
[8/12/2022 2:11:06 AM] Robbysan
[8/12/2022 2:11:02 AM] Ian
[8/12/2022 2:11:02 AM] Stan
[8/12/2022 2:11:02 AM] Robbysan
[8/12/2022 2:11:02 AM] Ian
[8/12/2022 2:11:02 AM] Stan
[8/12/2022 2:11:02 AM] Robbysan
[8/12/2022 2:11:02 AM] Ian
[8/12/2022 2:11:02 AM] Stan
[8/12/2022 2:11:02 AM] Robbysan
[8/12/2022 2:11:01 AM] Ian
[8/12/2022 2:11:01 AM] Stan
[8/12/2022 2:11:01 AM] Robbysan
[8/12/2022 2:11:01 AM] Ian
[8/12/2022 2:11:01 AM] Stan
[8/12/2022 2:11:01 AM] Robbysan
[8/12/2022 2:11:01 AM] Ian
[8/12/2022 2:11:01 AM] Stan
[8/12/2022 2:11:01 AM] Robbysan
[8/12/2022 2:11:01 AM] Ian
[8/12/2022 2:11:01 AM] Stan
[8/12/2022 2:11:01 AM] Robbysan
[8/12/2022 2:10:45 AM] Ian
[8/12/2022 2:10:45 AM] Stan
[8/12/2022 2:10:45 AM] Robbysan
[8/12/2022 2:10:16 AM] Ian
[8/12/2022 2:10:15 AM] Stan
[8/12/2022 2:10:15 AM] Robbysan
[8/12/2022 2:10:12 AM] Received STARTGAMEFINALIZE packet. Round init status: Started
[8/12/2022 2:10:12 AM] Task count: 0
[8/12/2022 2:10:12 AM] Received STARTGAMEFINALIZE packet. Round init status: WaitingForStartGameFinalize
[8/12/2022 2:10:08 AM] Generating an outpost for the start of the level... (Location: Cadmus Linea Mine, level type: Outpost)
[8/12/2022 2:10:08 AM] Level identifier: outpostlevel
[8/12/2022 2:09:57 AM] Ian
[8/12/2022 2:09:57 AM] Stan
[8/12/2022 2:09:57 AM] Robbysan
[8/12/2022 2:09:57 AM] Ian
[8/12/2022 2:09:57 AM] Stan
[8/12/2022 2:09:57 AM] Robbysan
[8/12/2022 2:09:57 AM] Ian
[8/12/2022 2:09:57 AM] Stan
[8/12/2022 2:09:57 AM] Robbysan
[8/12/2022 2:09:56 AM] Ian
[8/12/2022 2:09:56 AM] Stan
[8/12/2022 2:09:56 AM] Robbysan
[8/12/2022 2:09:56 AM] Ian
[8/12/2022 2:09:56 AM] Stan
[8/12/2022 2:09:56 AM] Robbysan
[8/12/2022 2:09:56 AM] Ian
[8/12/2022 2:09:56 AM] Stan
[8/12/2022 2:09:56 AM] Robbysan
[8/12/2022 2:09:56 AM] Ian
[8/12/2022 2:09:56 AM] Stan
[8/12/2022 2:09:56 AM] Robbysan
[8/12/2022 2:09:55 AM] Ian
[8/12/2022 2:09:55 AM] Stan
[8/12/2022 2:09:55 AM] Robbysan
[8/12/2022 2:09:55 AM] Ian
[8/12/2022 2:09:55 AM] Stan
[8/12/2022 2:09:55 AM] Robbysan
[8/12/2022 2:09:55 AM] Ian
[8/12/2022 2:09:55 AM] Stan
[8/12/2022 2:09:55 AM] Robbysan
[8/12/2022 2:08:54 AM] Ian
[8/12/2022 2:08:54 AM] Stan
[8/12/2022 2:08:54 AM] Robbysan
[8/12/2022 2:08:12 AM] Ian
[8/12/2022 2:08:12 AM] Stan
[8/12/2022 2:08:12 AM] Robbysan
[8/12/2022 2:08:07 AM] Ian
[8/12/2022 2:08:07 AM] Stan
[8/12/2022 2:08:07 AM] Robbysan
[8/12/2022 2:08:06 AM] Ian
[8/12/2022 2:08:06 AM] Stan
[8/12/2022 2:08:06 AM] Robbysan
[8/12/2022 2:08:06 AM] Ian
[8/12/2022 2:08:06 AM] Stan
[8/12/2022 2:08:06 AM] Robbysan
[8/12/2022 2:08:05 AM] Ian
[8/12/2022 2:08:05 AM] Stan
[8/12/2022 2:08:05 AM] Robbysan
[8/12/2022 2:08:05 AM] Ian
[8/12/2022 2:08:05 AM] Stan
[8/12/2022 2:08:05 AM] Robbysan
[8/12/2022 2:08:05 AM] Ian
[8/12/2022 2:08:05 AM] Stan
[8/12/2022 2:08:05 AM] Robbysan
[8/12/2022 2:08:05 AM] Ian
[8/12/2022 2:08:05 AM] Stan
[8/12/2022 2:08:05 AM] Robbysan
[8/12/2022 2:08:03 AM] Ian
[8/12/2022 2:08:03 AM] Stan
[8/12/2022 2:08:03 AM] Robbysan
[8/12/2022 2:08:03 AM] Ian
[8/12/2022 2:08:03 AM] Stan
[8/12/2022 2:08:03 AM] Robbysan
[8/12/2022 2:08:03 AM] Ian
[8/12/2022 2:08:03 AM] Stan
[8/12/2022 2:08:03 AM] Robbysan
[8/12/2022 2:01:41 AM] Ian
[8/12/2022 2:01:41 AM] Stan
[8/12/2022 2:01:40 AM] Robbysan
[8/12/2022 2:01:39 AM] Received STARTGAMEFINALIZE packet. Round init status: Started
[8/12/2022 2:01:34 AM] Task count: 0
[8/12/2022 2:01:34 AM] Received STARTGAMEFINALIZE packet. Round init status: WaitingForStartGameFinalize
[8/12/2022 2:01:31 AM] Generating an outpost for the end of the level... (Location: Cadmus Linea Mine, level type: LocationConnection)
[8/12/2022 2:01:29 AM] Generating an outpost for the start of the level... (Location: Huronia Biomedical Research Center, level type: LocationConnection)
[8/12/2022 2:01:25 AM] Level identifier: greatseabasic
[8/12/2022 2:01:16 AM] Ian
[8/12/2022 2:01:16 AM] Stan
[8/12/2022 2:01:16 AM] Robbysan
[8/12/2022 2:01:15 AM] Ian
[8/12/2022 2:01:15 AM] Stan
[8/12/2022 2:01:15 AM] Robbysan
[8/12/2022 2:01:15 AM] Ian
[8/12/2022 2:01:15 AM] Stan
[8/12/2022 2:01:15 AM] Robbysan
[8/12/2022 2:01:15 AM] WARNING: Failed to activate a location type change in the location "Drool". Location index out of bounds (2).
[8/12/2022 1:36:36 AM] Ian
[8/12/2022 1:36:33 AM] Stan
[8/12/2022 1:36:33 AM] Robbysan
[8/12/2022 1:36:31 AM] Received STARTGAMEFINALIZE packet. Round init status: Started
[8/12/2022 1:36:31 AM] Task count: 0
[8/12/2022 1:36:31 AM] Received STARTGAMEFINALIZE packet. Round init status: WaitingForStartGameFinalize
[8/12/2022 1:36:28 AM] Generating an outpost for the start of the level... (Location: Huronia Biomedical Research Center, level type: Outpost)
[8/12/2022 1:36:28 AM] Level identifier: outpostlevel
[8/12/2022 1:36:17 AM] Ian
```

**Comment by NilanthAnimosus** [comment]
Tested this in 0.18.15.1 - couldn't repro or cause any crashes with a few creatures I tried the Ahab's spear on. Noting I ran vanilla and this game had numerous mods active (and I'm not sure I would manage to find the correct mods or replicate similar conditions) I cannot imagine which mod might be relevant to causing this or if it just has a very low chance to occur.

I even tried killmonsters, severing limbs and removing the character (via console) with the rope attached without issues. the crash seems odd though, as it points at <DEV>\Barotrauma\BarotraumaShared\SharedSource\Items\Components\Rope.cs:line 0 which makes this a curiosity.
